### PR TITLE
Fix uninitialized variable append_ in class constructors

### PR DESCRIPTION
**Pull request by alexanderbobrovnik**
b/219048469

Test-On-Device: true

### CODE REVIEW

File: base/files/file.cc [hunk 1]
```diff
@@ -31,12 +31,24 @@ File::Info::~Info() = default;
 File::File()
     : error_details_(FILE_ERROR_FAILED),
       created_(false),
-      async_(false) {
+      async_(false)
+#if defined(STARBOARD)
+      ,
+      append_(false)
```
Comment (andrewsavage1) [inline base/files/file.cc:37]: Our footprint would be smaller if we defaulted `append_ = false` in the header and only set it in the copy constructors

**Review by andrewsavage1 — APPROVED** [review]

**Comment by TyHolc** [comment]
Rerunning the failed flows. For on-device tests it looks like some player_filter_tests on darwin-tvos/simulator/devel failed, but it doesn't seem related so I'll rerun those too.

**Comment by TyHolc** [comment]
> Rerunning the failed flows. For on-device tests it looks like some player_filter_tests on darwin-tvos/simulator/devel failed, but it doesn't seem related so I'll rerun those too.

Rerunning those and they failed again. If you think it's unrelated, we should probably rebase this and try again. Otherwise can you take a look.

Here's the failure message:

```
[test_runner.py:759:INFO]   FAILED TESTS:
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_aac_stereo_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc182 pointing to "beneath_the_canopy_aac_stereo.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_aac_stereo_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc182 pointing to "beneath_the_canopy_aac_stereo.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_aac_5_1_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc1a4 pointing to "beneath_the_canopy_aac_5_1.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_aac_5_1_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc1a4 pointing to "beneath_the_canopy_aac_5_1.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_aac_mono_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc1c3 pointing to "beneath_the_canopy_aac_mono.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_aac_mono_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc1c3 pointing to "beneath_the_canopy_aac_mono.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_opus_5_1_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc1e3 pointing to "beneath_the_canopy_opus_5_1.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_opus_5_1_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc1e3 pointing to "beneath_the_canopy_opus_5_1.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_opus_stereo_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc203 pointing to "beneath_the_canopy_opus_stereo.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_opus_stereo_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc203 pointing to "beneath_the_canopy_opus_stereo.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_opus_mono_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc226 pointing to "beneath_the_canopy_opus_mono.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/beneath_the_canopy_opus_mono_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc226 pointing to "beneath_the_canopy_opus_mono.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__beneath_the_canopy_aac_stereo_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc182 pointing to "beneath_the_canopy_aac_stereo.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__beneath_the_canopy_aac_5_1_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc1a4 pointing to "beneath_the_canopy_aac_5_1.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__beneath_the_canopy_aac_mono_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc1c3 pointing to "beneath_the_canopy_aac_mono.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__beneath_the_canopy_opus_5_1_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc1e3 pointing to "beneath_the_canopy_opus_5_1.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__beneath_the_canopy_opus_stereo_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc203 pointing to "beneath_the_canopy_opus_stereo.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__beneath_the_canopy_opus_mono_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc226 pointing to "beneath_the_canopy_opus_mono.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_329_ec3_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc247 pointing to "sintel_329_ec3.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__beneath_the_canopy_aac_stereo_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc182 pointing to "beneath_the_canopy_aac_stereo.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__beneath_the_canopy_aac_5_1_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc1a4 pointing to "beneath_the_canopy_aac_5_1.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__beneath_the_canopy_aac_mono_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc1c3 pointing to "beneath_the_canopy_aac_mono.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__beneath_the_canopy_opus_5_1_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc1e3 pointing to "beneath_the_canopy_opus_5_1.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__beneath_the_canopy_opus_stereo_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc203 pointing to "beneath_the_canopy_opus_stereo.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__beneath_the_canopy_opus_mono_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc226 pointing to "beneath_the_canopy_opus_mono.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__sintel_329_ec3_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc247 pointing to "sintel_329_ec3.dmp" }, false)
[test_runner.py:761:INFO]     AdaptiveAudioDecoderTests/AdaptiveAudioDecoderTest.SingleInput/sintel_381_ac3_dmp__to__sintel_381_ac3_dmp, where GetParam() = ({ 0x10e3dc25a pointing to "sintel_381_ac3.dmp", 0x10e3dc25a pointing to "sintel_381_ac3.dmp" }, false)
```

**Comment by alexanderbobrovnik** [comment]
Few checks are still failing after the rebase, seems unrelated to this change.

**Pull request merged**